### ISSUE TITLE
SSP Clear-text passwords on the client side during RDP

### ISSUE THREAD

**Issue by haim-n** [gentilkiwi/mimikatz#293]
Hi all,

When performing RDP and running mimikatz (sekurlsa::logonpasswords/ssp) on the originating client-side, during the RDP session, the clear-text is password is shown under "ssp".
It kind of makes sense, because the client should provide the password to the server, but it doesn't seem like a well-known vulnerability.
Why does it happen? Is it by design? Can it be prevented somehow?

Some notes:
1. The credential delegation setting 'Allow delegating default credentials' is turned off, as the default setting. The relevant registry values don't exist. Disabling it through gpedit doesn't make any difference. Enabling it causes clear-text password to be found under 'tspkg', as expected. So this is not related.
2. Not related to 'misc::memssp'.
3. Happens in any Windows version (10/2019/2016/2012R2).
4. Happens for local and domain users.
5. Happens even if the connecting user is part of the 'Protected Users' AD group.
6. Doesn't happen when performing RDP without NLA. Which makes sense, because the password isn't inputted on the client side... but NLA is considered a best practice for the last 15 years...
7. Doesn't happen when performing RDP with 'RestrictedAdmin' or 'Remote Credentials Guard', since the password isn't inputted by the user (SSO). But these options have some operational limitations and they don't allow to connect with a different user.

This is how it looks:
![ssp-clear](https://user-images.githubusercontent.com/45103390/85232158-bd2e8480-b405-11ea-9027-23b536c1e976.jpg)

Thanks a lot!

**Comment by christophschw** [comment]
Hey @haim-n 
did you find any informtion about this issue? I'm higly interested as well and your observations are very helpful.
Thanks

**Comment by haim-n** [comment]
Hi @christophschw
The only other information that I've found is that if the client computer has ‘Credential Guard’ turned on, than the password under "SSP" is stored in an encrypted manner and not shown in clear-text.

Other than that - no new information...
I didn't find anything documenting/describing this behavior on the internet. There is plenty of information about other vectors of gaining clear-text passwords (like WDigest, using 'misc::memssp' to bypass Cred Guard, abusing the 'delegating default credentials' setting), but I found nothing about what I've described above...

Dear @gentilkiwi - it would be awesome to get your insights! 🙂

**Comment by christophschw** [comment]
Hey @haim-n 

thank you for the reply.
I'm facing the exact same issue: A bunch of information regarding other SSP packages and their vulnerabilities, but not for this case.
Yeah, that would be really awesome @gentilkiwi 

All the best and stay safe
Christoph

**Comment by christophschw** [comment]
Follow the instruction :wink: 
https://github.com/gentilkiwi/mimikatz/wiki/howto-~-open-an-issue#use-cool-memes

![image](https://user-images.githubusercontent.com/13663130/103288825-134c3800-49e6-11eb-92bf-9f2d240dc463.png)

**Comment by haim-n** [comment]
@christophschw  Nice find!!! Thank you!


So if I will try to refer to my questions in the original post...:

> it doesn't seem like a well-known vulnerability.

Indeed it isn't well known! Benjamin called this an "underestimated category" and the only thing that we could find is his post from almost 8 years ago...

> Why does it happen? Is it by design?

Based on Google Translate, Benjamin says that: Our familiar process, LSASS, once again maintains a table of identification data that processes, providers, or drivers, can query to establish connections.

> Can it be prevented somehow?

Probably no, except for what I've written before (disabling NLA, RestrictedAdmin mode, Credential Guard).

**Comment by rjt** [comment]
When using strictly kerberos authentication,  is there still a password?   If so, does the password still show up in plaintext?

**Comment by haim-n** [comment]
@rjt yes, it makes no difference. Even if the user is inside the 'Protected Users' group, meaning that it cannot use NTLM, the password still show up in plaintext. I'm obviously RDPing into the FQDN instead of the IP to make it work over Kerberos.

**Comment by ruppde** [comment]
> Probably no, except for what I've written before (disabling NLA, RestrictedAdmin mode, Credential Guard).

Do I understand right that both the normal Credential Guard and Remote Credential Guard help?

Btw, Windows 2022 is still affected.

**Comment by haim-n** [comment]
> Do I understand right that both the normal Credential Guard and Remote Credential Guard help?

Normal Credential Guard helps.

As for Remote Credential Guard, this whole scenario doesn't really apply to it. When you RDP with Remote Credential Guard (or the 'legacy' RestrictedAdmin RDP), the authentication to the server will be done over Kerberos SSO, with the same credentials as the user the is currently connected to the source endpoint. It means that the credentials (NT Hash / TGT) of the privileged user (that has access to the server side) are stored in the client side anyway, and are at risk.
In a case where you connect to the endpoint with a low-privilege user, and RDP to a server with a high-privilege user, which is the scenario that I'm referring to in this behavior, Remote Credential Guard cannot be used.
